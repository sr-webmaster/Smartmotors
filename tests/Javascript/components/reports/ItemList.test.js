import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Component from '~/components/reports/ItemList.vue'
const FIXTURE_ITEMS = [
  {
    uuid: '1',
    deal_date: 'July 19, 2019',
    deal_number: '7878979878',
    type: 'Used',
    lot: 'Main Used',
    sales_rep: 'John McSalesman',
    customer: {
      id: '234525',
      name: 'John McSalesman'
    },
    sale_price: 234000,
    invoice_price: 987000
  }
]

const headers = [
  { text: 'Deal #', sortable: false, value: 'deal_number', align: 'left', class: 'font-weight-bold' },
  { text: 'Deal Written', sortable: false, value: 'deal_date', align: 'left', class: 'font-weight-bold' },
  { text: 'Type', sortable: false, value: 'type', align: 'left', class: 'font-weight-bold' },
  { text: 'Lot', sortable: false, value: 'lot', align: 'left', class: 'font-weight-bold' },
  { text: 'Sales rep', sortable: false, value: 'sales_rep', align: 'left', class: 'font-weight-bold' },
  { text: 'Customer', sortable: false, value: 'customer', align: 'left', class: 'font-weight-bold' },
  { text: 'Sale Price', sortable: false, value: 'sale_price', align: 'right', class: 'font-weight-bold' },
  { text: 'Invoice Price', sortable: false, value: 'invoice_price', align: 'right', class: 'font-weight-bold' },
  { text: 'Manage', sortable: false, value: 'manage', align: 'center' }
]

describe('ItemList', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('default', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue
      })
      expect(wrapper.element).toMatchSnapshot()
    })
    test('items set', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue,
        propsData: {
          items: FIXTURE_ITEMS,
          headers
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
    test('item is empty', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue,
        propsData: {
          items: [],
          headers
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Methods', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })
    test('onClickView() will emitted manage-view', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          items: FIXTURE_ITEMS,
          headers
        }
      })
      wrapper.vm.onClickView('123')
      expect(wrapper.emitted()['manage-view']).toBeTruthy()
      expect(wrapper.emitted()['manage-view'][0][0]).toEqual('123')
    })
  })
})
