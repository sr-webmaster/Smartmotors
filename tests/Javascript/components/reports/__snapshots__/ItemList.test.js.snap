// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList Snapshots default 1`] = `
<div>
  <div
    class="elevation-1"
  >
    <div
      class="v-table__overflow"
    >
      <table
        class="v-datatable v-table theme--light"
      >
        <thead>
          <tr />
          <tr
            class="v-datatable__progress"
          >
            <th
              class="column"
              colspan="0"
            />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="text-xs-center"
              colspan="0"
            >
              No matching records found
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
   
  <div
    class="layout align-center"
  >
    <div
      class="flex grow justify-center"
    >
      <div
        class="layout justify-center"
      >
        <ul
          class="v-pagination theme--light"
        >
          <li>
            <button
              class="v-pagination__navigation v-pagination__navigation--disabled"
              type="button"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
              >
                chevron_left
              </i>
            </button>
          </li>
          <li>
            <button
              class="v-pagination__item v-pagination__item--active primary"
              type="button"
            >
              1
            </button>
          </li>
          <li>
            <button
              class="v-pagination__navigation v-pagination__navigation--disabled"
              type="button"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
              >
                chevron_right
              </i>
            </button>
          </li>
        </ul>
      </div>
    </div>
     
    <div
      class="flex shrink"
    >
      <div
        class="v-input v-text-field v-select v-input--is-label-active v-input--is-dirty theme--light"
        role="combobox"
      >
        <div
          class="v-input__control"
        >
          <div
            class="v-input__slot"
          >
            <div
              class="v-select__slot"
            >
              <label
                aria-hidden="true"
                class="v-label v-label--active theme--light"
                style="left: 0px; position: absolute;"
              >
                Results Per Page
              </label>
              <div
                class="v-select__selections"
              >
                <div
                  class="v-select__selection v-select__selection--comma"
                >
                  10
                </div>
                <input
                  aria-label="Results Per Page"
                  aria-readonly="false"
                  autocomplete="on"
                  readonly="readonly"
                  type="text"
                />
              </div>
              <div
                class="v-input__append-inner"
              >
                <div
                  class="v-input__icon v-input__icon--append"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    arrow_drop_down
                  </i>
                </div>
              </div>
            </div>
            <div
              class="v-menu"
            >
              <div
                class="v-menu__content theme--light "
                style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
              >
                <div
                  class="v-select-list v-card theme--light"
                >
                  <div
                    class="v-list theme--light"
                    role="list"
                  >
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            5
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      class="primary--text"
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            10
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            15
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            25
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            30
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            50
                          </div>
                        </div>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="v-text-field__details"
          >
            <div
              class="v-messages theme--light"
            >
              <div
                class="v-messages__wrapper"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ItemList Snapshots item is empty 1`] = `
<div>
  <div
    class="elevation-1"
  >
    <div
      class="v-table__overflow"
    >
      <table
        class="v-datatable v-table theme--light"
      >
        <thead>
          <tr />
          <tr
            class="v-datatable__progress"
          >
            <th
              class="column"
              colspan="0"
            />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="text-xs-center"
              colspan="0"
            >
              No matching records found
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
   
  <div
    class="layout align-center"
  >
    <div
      class="flex grow justify-center"
    >
      <div
        class="layout justify-center"
      >
        <ul
          class="v-pagination theme--light"
        >
          <li>
            <button
              class="v-pagination__navigation v-pagination__navigation--disabled"
              type="button"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
              >
                chevron_left
              </i>
            </button>
          </li>
          <li>
            <button
              class="v-pagination__item v-pagination__item--active primary"
              type="button"
            >
              1
            </button>
          </li>
          <li>
            <button
              class="v-pagination__navigation v-pagination__navigation--disabled"
              type="button"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
              >
                chevron_right
              </i>
            </button>
          </li>
        </ul>
      </div>
    </div>
     
    <div
      class="flex shrink"
    >
      <div
        class="v-input v-text-field v-select v-input--is-label-active v-input--is-dirty theme--light"
        role="combobox"
      >
        <div
          class="v-input__control"
        >
          <div
            class="v-input__slot"
          >
            <div
              class="v-select__slot"
            >
              <label
                aria-hidden="true"
                class="v-label v-label--active theme--light"
                style="left: 0px; position: absolute;"
              >
                Results Per Page
              </label>
              <div
                class="v-select__selections"
              >
                <div
                  class="v-select__selection v-select__selection--comma"
                >
                  10
                </div>
                <input
                  aria-label="Results Per Page"
                  aria-readonly="false"
                  autocomplete="on"
                  readonly="readonly"
                  type="text"
                />
              </div>
              <div
                class="v-input__append-inner"
              >
                <div
                  class="v-input__icon v-input__icon--append"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    arrow_drop_down
                  </i>
                </div>
              </div>
            </div>
            <div
              class="v-menu"
            >
              <div
                class="v-menu__content theme--light "
                style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
              >
                <div
                  class="v-select-list v-card theme--light"
                >
                  <div
                    class="v-list theme--light"
                    role="list"
                  >
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            5
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      class="primary--text"
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            10
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            15
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            25
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            30
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            50
                          </div>
                        </div>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="v-text-field__details"
          >
            <div
              class="v-messages theme--light"
            >
              <div
                class="v-messages__wrapper"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ItemList Snapshots items set 1`] = `
<div>
  <div
    class="elevation-1"
  >
    <div
      class="v-table__overflow"
    >
      <table
        class="v-datatable v-table theme--light"
      >
        <thead>
          <tr />
          <tr
            class="v-datatable__progress"
          >
            <th
              class="column"
              colspan="0"
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
   
  <div
    class="layout align-center"
  >
    <div
      class="flex grow justify-center"
    >
      <div
        class="layout justify-center"
      >
        <ul
          class="v-pagination theme--light"
        >
          <li>
            <button
              class="v-pagination__navigation v-pagination__navigation--disabled"
              type="button"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
              >
                chevron_left
              </i>
            </button>
          </li>
          <li>
            <button
              class="v-pagination__item v-pagination__item--active primary"
              type="button"
            >
              1
            </button>
          </li>
          <li>
            <button
              class="v-pagination__navigation v-pagination__navigation--disabled"
              type="button"
            >
              <i
                aria-hidden="true"
                class="v-icon material-icons theme--light"
              >
                chevron_right
              </i>
            </button>
          </li>
        </ul>
      </div>
    </div>
     
    <div
      class="flex shrink"
    >
      <div
        class="v-input v-text-field v-select v-input--is-label-active v-input--is-dirty theme--light"
        role="combobox"
      >
        <div
          class="v-input__control"
        >
          <div
            class="v-input__slot"
          >
            <div
              class="v-select__slot"
            >
              <label
                aria-hidden="true"
                class="v-label v-label--active theme--light"
                style="left: 0px; position: absolute;"
              >
                Results Per Page
              </label>
              <div
                class="v-select__selections"
              >
                <div
                  class="v-select__selection v-select__selection--comma"
                >
                  10
                </div>
                <input
                  aria-label="Results Per Page"
                  aria-readonly="false"
                  autocomplete="on"
                  readonly="readonly"
                  type="text"
                />
              </div>
              <div
                class="v-input__append-inner"
              >
                <div
                  class="v-input__icon v-input__icon--append"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    arrow_drop_down
                  </i>
                </div>
              </div>
            </div>
            <div
              class="v-menu"
            >
              <div
                class="v-menu__content theme--light "
                style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
              >
                <div
                  class="v-select-list v-card theme--light"
                >
                  <div
                    class="v-list theme--light"
                    role="list"
                  >
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            5
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      class="primary--text"
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            10
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            15
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            25
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            30
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      role="listitem"
                    >
                      <a
                        class="v-list__tile v-list__tile--link theme--light"
                      >
                        <div
                          class="v-list__tile__content"
                        >
                          <div
                            class="v-list__tile__title"
                          >
                            50
                          </div>
                        </div>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="v-text-field__details"
          >
            <div
              class="v-messages theme--light"
            >
              <div
                class="v-messages__wrapper"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
